src/components/carousel/index.tsx

```diff
@@ -52,28 +52,27 @@ export const Carousel: FunctionComponent<CarouselProps> = (userProps: CarouselPr
 	const autoSwipeTimer = useRef<number>();
 	const isNavigation = typeof props.navigation === 'function';
 
-	if (props.dynamic) {
-		useEffect(() => {
-			const newItems = updateNodes(
-				itemsRef.current,
-				props.children,
-				prevChildren,
-				props.slide,
-				props.infinite,
-			);
+	useEffect(() => {
+		if (props.dynamic) return 
+		const newItems = updateNodes(
+			itemsRef.current,
+			props.children,
+			prevChildren,
+			props.slide,
+			props.infinite,
+		);
 
-			setItems(newItems);
-			itemsRef.current = newItems;
-			if (
-				page < props.pageCount &&
-				prevChildren &&
-				prevChildren?.length < props.children.length
-			) {
-				slide(SlideDirection.Right);
-				setPage(page + 1);
-			}
-		}, [props.children]);
-	}
+		setItems(newItems);
+		itemsRef.current = newItems;
+		if (
+			page < props.pageCount &&
+			prevChildren &&
+			prevChildren?.length < props.children.length
+		) {
+			slide(SlideDirection.Right);
+			setPage(page + 1);
```

Comment:
this isn't guaranteed either, if you want to use the previous state you should pass a callback like so...

`setPage((previousPage) => previousPage + 1))`